scripts/specs.ts

```diff
@@ -1,18 +1,54 @@
+import assert from "node:assert/strict";
+
 import webSpecs from 'web-specs' assert { type: 'json' };
 
 import features from '../index.js';
 
-const specUrls = new Set(webSpecs.map(spec => spec.nightly.url));
+const specUrls = webSpecs.map(spec => new URL(spec.nightly.url));
+
+type exception = [url: string, message: string];
+const exceptions: exception[] = [
+    [
+        "https://wicg.github.io/navigation-api/",
+        "This spec is moving to WHATWG HTML. Remove this exception when https://github.com/whatwg/html/pull/8502 merges."
+    ],
+];
+
+function isOK(url: URL, allowlist: exception[] = exceptions) {
+    for (const specUrl of specUrls) {
+        if (specUrl.protocol !== url.protocol || specUrl.origin !== url.origin) {
```

Comment:
The origin includes the scheme, so this can be simplified:

```suggestion
        if (specUrl.origin !== url.origin) {
```